gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/matching/common/statistics/GraphStatisticsDataSink.java

```diff
@@ -0,0 +1,224 @@
+ * The statistics are currently restricted to the ones used by the cost plan
+ * estimator of the cypher pattern matching operator {@link CypherPatternMatching}.
+ * Caution: This data sink ONLY knows how to process {@link LogicalGraph}s!
+ */
+public class GraphStatisticsDataSink implements DataSink {
+
+  /**
+   * Directory/HDFS directory name for statistics output
+   */
+  private final String path;
+
+  /**
+   * Default constructor
+   * @param path file path in a local filesystem or HDFS
+   */
+  public GraphStatisticsDataSink(final String path) {
+    this.path = path;
+  }
+  /* (non-Javadoc)
+   * @see org.gradoop.flink.io.api.DataSink#write(org.gradoop.flink.model.api.epgm.LogicalGraph)
+   */
+  @Override
+  public void write(final LogicalGraph logicalGraph) throws IOException {
+    write(logicalGraph, false);
+  }
+
+
+  /* (non-Javadoc)
+   * @see org.gradoop.flink.io.api.DataSink#write(org.gradoop.flink.model.api.epgm.LogicalGraph, boolean)
+   */
+  @Override
+  public void write(final LogicalGraph graph, final boolean overwrite) throws IOException {
+
+    // VertexCountRunner.main(args)
+    writeCSV(
+        new VertexCount()
+        .execute(graph)
+        .map(new ObjectTo1<>()),
+        GraphStatisticsReader.FILE_VERTEX_COUNT,
+        overwrite);
+    // EdgeCountRunner.main(args)
+    writeCSV(
+        new EdgeCount()
+        .execute(graph)
+        .map(new ObjectTo1<>()),
+        GraphStatisticsReader.FILE_EDGE_COUNT,
+        overwrite);
+    // VertexLabelDistributionRunner.main(args)
+    writeCSV(
+        new VertexLabelDistribution()
+        .execute(graph),
+        GraphStatisticsReader.FILE_VERTEX_COUNT_BY_LABEL,
+        overwrite);
+    //    EdgeLabelDistributionRunner.main(args);
+    writeCSV(
+        new EdgeLabelDistribution()
+        .execute(graph) ,
+        GraphStatisticsReader.FILE_EDGE_COUNT_BY_LABEL,
+        overwrite);
+    //    DistinctSourceVertexCountRunner.main(args);
+    writeCSV(
+        new DistinctSourceIds()
+        .execute(graph)
+        .map(new ObjectTo1<>()),
+        GraphStatisticsReader.FILE_DISTINCT_SOURCE_VERTEX_COUNT,
+        overwrite);
+    //    DistinctTargetVertexCountRunner.main(args);
+    writeCSV(
+        new DistinctTargetIds()
+        .execute(graph)
+        .map(new ObjectTo1<>()),
+        GraphStatisticsReader.FILE_DISTINCT_TARGET_VERTEX_COUNT,
+        overwrite);
+    //    DistinctSourceVertexCountByEdgeLabelRunner.main(args);
+    writeCSV(
+        new DistinctSourceIdsByEdgeLabel()
+        .execute(graph),
+        GraphStatisticsReader.FILE_DISTINCT_SOURCE_VERTEX_COUNT_BY_EDGE_LABEL,
+        overwrite);
+    //    DistinctTargetVertexCountByEdgeLabelRunner.main(args);
+    writeCSV(
+        new DistinctTargetIdsByEdgeLabel()
+        .execute(graph),
+        GraphStatisticsReader.FILE_DISTINCT_TARGET_VERTEX_COUNT_BY_EDGE_LABEL,
+        overwrite);
+    //    SourceAndEdgeLabelDistributionRunner.main(args);
+    writeCSV(
```

Comment:
I think the formatting is a bit confusing. Should be something like
`writeCSV(
\t new SourceLabelAndEdgeLabelDistribution()
\t\t  .execute(graph)
\t\t  .map(value -> Tuple3.of(value.f0.f0, value.f0.f1, value.f1))
\t\t  .returns(new TypeHint<Tuple3<String, String, Long>>() { }),
\t GraphStatisticsReader.FILE_EDGE_COUNT_BY_SOURCE_VERTEX_AND_EDGE_LABEL,
\t overwrite);`

For all occurences of the writeCSV call. Furthermore I'm not sure whether we have this convention but personally I would prefer the method name `writeCsv`.